components/gitpod-protocol/src/protocol.ts

```diff
@@ -273,7 +274,19 @@ export interface AdditionalUserData extends Partial<WorkspaceTimeoutSetting> {
     // whether the user has been migrated to team attribution.
     // a corresponding feature flag (team_only_attribution) triggers the migration.
     isMigratedToTeamOnlyAttribution?: boolean;
+
+    // workspace creation options per clone URL
+    workspaceCreationOptionsPerCloneURL?: WorkspaceCreationOptionWithCloneURL[];
```

Comment:
If nothing else, what if we store it indexed by the cloneURL instead of a list? It would at least avoid linear scans of the list on lookup and more easily handles duplicates (by not allowing them)

Reply:
Over the course of repos I work on, I may toggle that option on many of them. Some for my organisational work, some for personal.

I'll leave it up to you, but it feels odd to store this much information in an extra user blob

Reply:
I believe it is very unlikely that someone would remember options for more than say 20 repos but in case this happens, I'll code to forget the oldest entries.